jans-cedarling/cedarling/src/authz/mod.rs

```diff
@@ -272,75 +253,58 @@ impl Authz {
-                    &tokens_metadata.id_tokens.claim_mapping)
-                    .map_err(AuthorizeError::CreateIdTokenEntity)?,
-            )
-            // Add userinfo_token entity
-            .userinfo_token(
-                create_userinfo_token_entity(auth_conf.mapping_userinfo_token.as_deref(), 
-                    policy_store, 
-                    &tokens.userinfo_token, 
-                    &tokens_metadata.userinfo_tokens.claim_mapping)
-                .map_err(AuthorizeError::CreateUserinfoTokenEntity)?
-            )
-            // Add User entity
-            .user_entity(
-                create_user_entity(auth_conf.mapping_user.as_deref(),
+        
+        // build workload entity
+        let workload = create_workload_entity(
+                    auth_conf.mapping_workload.as_deref(),
                     policy_store,
                     tokens,
-                    // parents for Jans::User entity
-                    HashSet::from_iter(role_entities.iter().map(|e|e.uid())),
-                    trusted_issuer
-                )
-                .map_err(AuthorizeError::CreateUserEntity)?,
-            )
-            // Add an entity created from the resource in the request
-            .resource_entity(create_resource_entity(
+                )?;
+
+        // build role entity
+        let role = create_role_entities(policy_store, tokens)?;
+    
+        // build user entity
+        let user = create_user_entity(auth_conf.mapping_user.as_deref(),
+                policy_store,
+                tokens,
+                HashSet::from_iter(role.iter().map(|e| e.uid())))?;
+    
+        // build access_token Entity
+        let access_token = tokens.access_token.as_ref().map(|tkn| {
+             create_access_token(auth_conf.mapping_access_token.as_deref(), policy_store, tkn)
+                .map_err(AuthorizeError::CreateAccessTokenEntity)
+        }).transpose()?;
+
+        // build id_token Entity
+        let id_token = tokens.id_token.as_ref().map(|tkn| {
+             create_id_token_entity(auth_conf.mapping_id_token.as_deref(), policy_store, tkn)
+                .map_err(AuthorizeError::CreateIdTokenEntity)
+        }).transpose()?;
+
+        // build userinfo_token Entity
+        let userinfo_token = tokens.userinfo_token.as_ref().map(|tkn| {
+             create_userinfo_token_entity(auth_conf.mapping_userinfo_token.as_deref(), policy_store, tkn)
+                .map_err(AuthorizeError::CreateUserinfoTokenEntity)
+        }).transpose()?;
+        
+        // build resource entity
+        let resource = create_resource_entity(
                 request.resource.clone(),
                 &self.config.policy_store.schema.json,
-            )?)
-            // Add Role entities
-            .role_entities(role_entities);
+            )?;
 
-        Ok(data.build())
+        Ok(AuthorizeEntitiesData { workload, access_token, id_token, userinfo_token, user, resource, role })
```

Comment:
Why you decided to avoid using builder?

Reply:
it will be more readable to use 
```
Ok(AuthorizeEntitiesData { 
            // build workload entity
            workload: create_workload_entity(
                    auth_conf.mapping_workload.as_deref(),
                    policy_store,
                    tokens,
                )?,
            ...
```

Reply:
the old implementation was not readable at all.

Reply:
i just split it into multiple lines.